packages/middleware-endpoint/src/adaptors/getEndpointFromInstructions.ts

```diff
@@ -36,6 +38,13 @@ export const getEndpointFromInstructions = async <
   clientConfig: Partial<EndpointResolvedConfig<T>> & Config,
   context?: HandlerExecutionContext
 ): Promise<EndpointV2> => {
+  if (!clientConfig.endpoint) {
+    const endpointFromConfig = await getEndpointFromConfig(clientConfig.serviceId || "");
+    if (endpointFromConfig) {
+      clientConfig.endpoint = () => Promise.resolve(toEndpointV1(endpointFromConfig));
+    }
+  }
```

Comment:
Would checking the config files on each operation invocation be expensive for I/O?

Maybe this should be in a `resolve*Config` function since it shouldn't change between calls.

Reply:
No. We cache the contents of the config file in https://github.com/awslabs/smithy-typescript/blob/main/packages/shared-ini-file-loader/src/slurpFile.spec.ts, and read operation is called only once per file.